src/org/zaproxy/zap/extension/api/CoreAPI.java

```diff
@@ -838,9 +843,21 @@ public void process(Alert alert) {
 			processAlerts(
 					this.getParam(params, PARAM_BASE_URL, (String) null), 
 					this.getParam(params, PARAM_START, -1), 
-					this.getParam(params, PARAM_COUNT, -1), counter);
+					this.getParam(params, PARAM_COUNT, -1), 
+					this.getParam(params, PARAM_RISK, -1), counter);
 			
 			result = new ApiResponseElement(name, Integer.toString(counter.getCount()));
+		} else if (VIEW_ALERTS_SUMMARY.equals(name)) {
+			final ApiResponseList resultList = new ApiResponseList(name);
+			Map<String, Object> alertData = new HashMap<>();
+			int[] riskIdArray = {Alert.RISK_HIGH, Alert.RISK_MEDIUM, Alert.RISK_LOW, Alert.RISK_INFO};
+			CounterProcessor<Alert> counter = new CounterProcessor<>();
+			for (int riskId:riskIdArray) {
+				processAlerts(this.getParam(params, PARAM_BASE_URL, (String) null), -1, -1, riskId, counter);
```

Comment:
Right, "4 times" less processing (less GC pressure, less I/O ops...) and it's not more complicated, it just moves the logic around into the processor. For example (if you don't want to have the processor create the map for you) it could be:
```Java
final int[] risks = { 0, 0, 0, 0 };
Processor<Alert> counter = new Processor<Alert>() {

    @Override
    public void process(Alert alert) {
        risks[alert.getRisk()]++;
    }
};
processAlerts(this.getParam(params, PARAM_BASE_URL, (String) null), -1, -1, -1, counter);

Map<String, Integer> alertData = new HashMap<>();
for (int i = 0; i < risks.length; i++) {
    alertData.put(Alert.MSG_RISK[i], risks[i]);
}

// ...
```